staging/src/k8s.io/apiserver/pkg/storage/cacher/cache_watcher.go

```diff
@@ -57,7 +57,7 @@ type cacheWatcher struct {
 	done      chan struct{}
 	filter    filterWithAttrsFunc
 	stopped   bool
-	forget    func(bool)
+	forget    func(bool, bool)
```

Comment:
instead of changing this all-over-the-place, I would prefer to pass some callback like `onUnresponsiveReceiver` during cacheWatcher creation.
It should be much simpler to plug this in.